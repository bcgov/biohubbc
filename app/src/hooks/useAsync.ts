import { useRef } from 'react';

export type AsyncFunction<Q extends any[], R> = (...args: Q) => Promise<R>;

export type WrappedAsyncFunction<Q extends any[], S> = (...args: Q) => Promise<S>;

/**
 * Wraps an async function to prevent duplicate calls if the previous call is still pending.
 *
 * Example:
 *
 * ```
 * const myAsyncFunction = useAsync(
 *   (param: number) => asyncFunction(param1)
 * )
 *
 * await myAsyncFunction(1) // calls `asyncFunction`, where param=1
 * (call 1 is pending)
 * await myAsyncFunction(2) // returns promise from call 1
 * (call 1 is fulfilled)
 * await myAsyncFunction(2) // calls `asyncFunction`, where param=2
 * (call 2 is pending)
 * ```
 *
 * @template Q
 * @template R
 * @param {AsyncFunction<Q, R>} asyncFunction the async function to wrap
 * @param {IUseAsyncOptions} [options]
 * @return {*}  {WrappedAsyncFunction<Q, R>}
 */
export const useAsync = <Q extends any[], R>(asyncFunction: AsyncFunction<Q, R>): WrappedAsyncFunction<Q, R> => {
  const ref = useRef<Promise<R>>();

  const isPending = useRef(false);

  const wrappedAsyncFunction: WrappedAsyncFunction<Q, R> = async (...args) => {
    if (ref.current && isPending.current) {
      return ref.current;
    }

    isPending.current = true;

    ref.current = asyncFunction(...args)
      .then(
        (response: R) => {
          isPending.current = false;

          return response;
        },
        (error) => {
          isPending.current = false;

          throw error;
        }
      )
      .catch((error) => {
        isPending.current = false;

        throw error;
      });

    return ref.current;
  };

  return wrappedAsyncFunction;
};
