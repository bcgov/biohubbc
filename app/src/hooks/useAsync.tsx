import { useRef } from 'react';

export type AsyncFunction<AFArgs extends any[], AFResponse> = (...args: AFArgs) => Promise<AFResponse>;

/**
 * Wraps an async function to prevent duplicate calls if the previous call is still pending.
 *
 * Example:
 *
 * ```
 * const myAsyncFunction = useAsync(
 *   (param: number) => asyncFunction(param1)
 * )
 *
 * await myAsyncFunction(1) // calls `asyncFunction`, where param=1
 * // call 1 is pending
 * await myAsyncFunction(2) // returns pending promise from call 1
 * await myAsyncFunction(2) // returns pending promise from call 1
 * // call 1 is fulfilled
 * await myAsyncFunction(2) // calls `asyncFunction`, where param=2
 * // call 2 is pending
 * ```
 *
 * @template AFArgs `AsyncFunction` argument types.
 * @template AFResponse `AsyncFunction` response type.
 * @param {AsyncFunction<AFArgs, AFResponse>} asyncFunction the async function to wrap
 * @return {*}  {AsyncFunction<AFArgs, AFResponse>}
 */
export const useAsync = <AFArgs extends any[], AFResponse>(
  asyncFunction: AsyncFunction<AFArgs, AFResponse>
): AsyncFunction<AFArgs, AFResponse> => {
  const ref = useRef<Promise<AFResponse>>();

  const isPending = useRef(false);

  const wrappedAsyncFunction: AsyncFunction<AFArgs, AFResponse> = async (...args) => {
    if (ref.current && isPending.current) {
      return ref.current;
    }

    isPending.current = true;

    ref.current = asyncFunction(...args).then(
      (response: AFResponse) => {
        isPending.current = false;

        return response;
      },
      (error) => {
        isPending.current = false;

        throw error;
      }
    );

    return ref.current;
  };

  return wrappedAsyncFunction;
};
