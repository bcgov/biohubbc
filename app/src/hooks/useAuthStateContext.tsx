import { useContext } from 'react';
import { AuthStateContext, IAuthState } from '../contexts/authStateContext';

/**
 * Returns an instance of `IAuthState` from `AuthStateContext`.
 *
 * @return {*}  {IAuthState}
 */
export const useAuthStateContext = (): IAuthState => {
  const context = useContext(AuthStateContext);

  if (!context) {
    throw Error(
      'AuthStateContext is undefined, please verify you are calling useAuthStateContext() as child of an <AuthStateProvider> component.'
    );
  }

  return context;
};
