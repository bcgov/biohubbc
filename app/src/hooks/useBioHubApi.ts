import { useKeycloak } from '@react-keycloak/web';
import axios from 'axios';
import { IProject, IProjectPostObject } from 'interfaces/project-interfaces';
import { IActivity, ICreateActivity, ICreateProjectResponse, ITemplate } from 'interfaces/useBioHubApi-interfaces';
import moment from 'moment';
import { useMemo } from 'react';

const API_HOST = process.env.REACT_APP_API_HOST;
const API_PORT = process.env.REACT_APP_API_PORT;

const API_URL = (API_PORT && `${API_HOST}:${API_PORT}`) || API_HOST || 'api-dev-biohubbc.apps.silver.devops.gov.bc.ca';

/**
 * Returns an instance of axios with baseURL and authorization headers set.
 *
 * @return {*}
 */
const useApi = () => {
  const { keycloak } = useKeycloak();
  const instance = useMemo(() => {
    return axios.create({
      headers: {
        Authorization: `Bearer ${keycloak?.token}`
      },
      baseURL: API_URL
    });
  }, [keycloak]);

  return instance;
};

/**
 * Returns a set of supported api methods.
 *
 * @return {object} object whose properties are supported api methods.
 */
export const useBiohubApi = () => {
  const api = useApi();

  /**
   * Get all projects.
   *
   * @return {*}  {Promise<IProject[]>}
   */
  const getProjects = async (): Promise<IProject[]> => {
    // const { data } = await api.get(`/api/projects`);

    // return data;

    // TODO: stub for development
    return [
      {
        id: 1,
        name: 'Project Name 1',
        objectives: 'Project Objectives 1',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 1',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 1',
        caveats: 'Caveats 1',
        comments: 'Comments 1'
      },
      {
        id: 2,
        name: 'Project Name 2',
        objectives: 'Project Objectives 2',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 2',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 2',
        caveats: 'Caveats 2',
        comments: 'Comments 2'
      },
      {
        id: 3,
        name: 'Project Name 3',
        objectives: 'Project Objectives 3',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 3',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 3',
        caveats: 'Caveats 3',
        comments: 'Comments 3'
      },
      {
        id: 4,
        name: 'Project Name 4',
        objectives: 'Project Objectives 4',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 4',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 4',
        caveats: 'Caveats 4',
        comments: 'Comments 4'
      }
    ];
  };

  /**
   * Get a project based on its ID.
   *
   * @param {projectId} projectId
   * @return {*}  {Promise<IProject>}
   */
  const getProject = async (projectId: number): Promise<IProject> => {
    const { data } = await api.get(`/api/project/${projectId}`);

    return data;
  };

  /**
   * Create a new project.
   *
   * @param {IProjectPostObject} project
   * @return {*}  {Promise<ICreateProjectResponse>}
   */
  const createProject = async (project: IProjectPostObject): Promise<ICreateProjectResponse> => {
    const { data } = await api.post('/api/project', project);

    return data;
  };

  /**
   * Get a template based on its ID.
   *
   * @param {templateId} templateId
   * @return {*}  {Promise<ITemplate>}
   */
  const getTemplate = async (templateId: number): Promise<ITemplate> => {
    const { data } = await api.get(`/api/template/${templateId}`);

    return data;
  };

  /**
   * Create a new activity record.
   *
   * @param {ICreateActivity} activity
   * @return {*}  {Promise<IActivity>}
   */
  const createActivity = async (activity: ICreateActivity): Promise<IActivity> => {
    const { data } = await api.post('/api/activity', activity);

    return data;
  };

  /**
   * Fetch all code sets.
   *
   * @return {*}  {Promise<any>}
   */
  const getAllCodes = async (): Promise<any> => {
    const { data } = await api.get('/api/codes/');

    return data;
  };

  /**
   * Fetch the api json-schema spec.
   *
   * @return {*}  {Promise<any>}
   */
  const getApiSpec = async (): Promise<any> => {
    const { data } = await api.get('/api/api-docs/');

    return data;
  };

  return {
    getProjects,
    getProject,
    getTemplate,
    createProject,
    createActivity,
    getAllCodes,
    getApiSpec
  };
};
