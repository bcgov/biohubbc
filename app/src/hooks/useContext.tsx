import { CodesContext, ICodesContext } from 'contexts/codesContext';
import { ConfigContext, IConfig } from 'contexts/configContext';
import { DialogContext, IDialogContext } from 'contexts/dialogContext';
import { IObservationsContext, ObservationsContext } from 'contexts/observationsContext';
import { IObservationsTableContext, ObservationsTableContext } from 'contexts/observationsTableContext';
import { IProjectContext, ProjectContext } from 'contexts/projectContext';
import { ISurveyContext, SurveyContext } from 'contexts/surveyContext';
import { ITaxonomyContext, TaxonomyContext } from 'contexts/taxonomyContext';
import { useContext } from 'react';

/**
 * Returns an instance of `IConfig` from `ConfigContext`.
 *
 * @return {*}  {IConfig}
 */
export const useConfigContext = (): IConfig => {
  const context = useContext(ConfigContext);

  if (!context) {
    throw Error(
      'ConfigContext is undefined, please verify you are calling useConfigContext() as child of an <ConfigContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `ICodesContext` from `CodesContext`.
 *
 * @return {*}  {ICodesContext}
 */
export const useCodesContext = (): ICodesContext => {
  const context = useContext(CodesContext);

  if (!context) {
    throw Error(
      'CodesContext is undefined, please verify you are calling useCodesContext() as child of an <CodesContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `IDialogContext` from `DialogContext`.
 *
 * @return {*}  {IDialogContext}
 */
export const useDialogContext = (): IDialogContext => {
  const context = useContext(DialogContext);

  if (!context) {
    throw Error(
      'DialogContext is undefined, please verify you are calling useDialogContext() as child of an <DialogContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `IProjectContext` from `ProjectContext`.
 *
 * @return {*}  {IProjectContext}
 */
export const useProjectContext = (): IProjectContext => {
  const context = useContext(ProjectContext);

  if (!context) {
    throw Error(
      'ProjectContext is undefined, please verify you are calling useProjectContext() as child of an <ProjectContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `ISurveyContext` from `SurveyContext`.
 *
 * @return {*}  {ISurveyContext}
 */
export const useSurveyContext = (): ISurveyContext => {
  const context = useContext(SurveyContext);

  if (!context) {
    throw Error(
      'SurveyContext is undefined, please verify you are calling useSurveyContext() as child of an <SurveyContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `IObservationsContext` from `SurveyContext`.
 *
 * @return {*}  {IObservationsContext}
 */
export const useObservationsContext = (): IObservationsContext => {
  const context = useContext(ObservationsContext);

  if (!context) {
    throw Error(
      'ObservationsContext is undefined, please verify you are calling useObservationsContext() as child of an <ObservationsContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `IObservationsTableContext` from `SurveyContext`.
 *
 * @return {*}  {IObservationsTableContext}
 */
export const useObservationsTableContext = (): IObservationsTableContext => {
  const context = useContext(ObservationsTableContext);

  if (!context) {
    throw Error(
      'ObservationsTableContext is undefined, please verify you are calling useObservationsTableContext() as child of an <ObservationsTableContextProvider> component.'
    );
  }

  return context;
};

/**
 * Returns an instance of `ITaxonomyContext` from `SurveyContext`.
 *
 * @return {*}  {ITaxonomyContext}
 */
export const useTaxonomyContext = (): ITaxonomyContext => {
  const context = useContext(TaxonomyContext);

  if (!context) {
    throw Error(
      'TaxonomyContext is undefined, please verify you are calling useTaxonomyContext() as child of an <TaxonomyContextProvider> component.'
    );
  }

  return context;
};
