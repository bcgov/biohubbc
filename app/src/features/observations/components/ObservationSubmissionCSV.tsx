import Box from '@material-ui/core/Box';
import CircularProgress from '@material-ui/core/CircularProgress';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Tab from '@material-ui/core/Tab';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';
import Tabs from '@material-ui/core/Tabs';
import Typography from '@material-ui/core/Typography';
import { useBiohubApi } from 'hooks/useBioHubApi';
import { IGetSubmissionCSVForViewItem, IGetSubmissionCSVForViewResponse } from 'interfaces/useObservationApi.interface';
import React, { useCallback, useEffect, useState } from 'react';
import { useParams } from 'react-router';
import { handleChangePage, handleChangeRowsPerPage } from 'utils/tablePaginationUtils';

const useStyles = makeStyles({
  table: {
    minWidth: 650
  },
  heading: {
    fontWeight: 'bold'
  },
  tableCellBorderTop: {
    borderTop: '1px solid rgba(224, 224, 224, 1)'
  }
});

const a11yProps = (index: number) => {
  return {
    id: `full-width-tab-${index}`,
    'aria-controls': `full-width-tabpanel-${index}`
  };
};

export interface ITabPanelProps {
  children: any;
  index: any;
  value: any;
}

const TabPanel: React.FC<ITabPanelProps> = (props) => {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`full-width-tabpanel-${index}`}
      aria-labelledby={`full-width-tab-${index}`}
      {...other}>
      {value === index && (
        <Box p={3}>
          <Typography component="div">{children}</Typography>
        </Box>
      )}
    </div>
  );
};

export interface IObservationSubmissionCSVProps {
  submissionId: number;
}

const ObservationSubmissionCSV: React.FC<IObservationSubmissionCSVProps> = (props) => {
  const biohubApi = useBiohubApi();
  const classes = useStyles();
  const urlParams = useParams();

  const { submissionId } = props;

  const [rowsPerPage, setRowsPerPage] = useState(50);
  const [page, setPage] = useState(0);
  const [value, setValue] = useState(0);
  const [isLoadingSubmissionCSV, setIsLoadingSubmissionCSV] = useState(true);
  const [submissionCSVDetails, setSubmissionCSVDetails] = useState<IGetSubmissionCSVForViewResponse | null>(null);

  const getSubmissionCSVDetails = useCallback(async () => {
    const submissionCSVWithDetailsResponse = await biohubApi.observation.getSubmissionCSVForView(
      urlParams['id'],
      urlParams['survey_id'],
      submissionId
    );

    if (!submissionCSVWithDetailsResponse) {
      return;
    }

    setSubmissionCSVDetails(submissionCSVWithDetailsResponse);
  }, [biohubApi.observation, urlParams, submissionId]);

  useEffect(() => {
    if (isLoadingSubmissionCSV && !submissionCSVDetails) {
      getSubmissionCSVDetails();
      setIsLoadingSubmissionCSV(false);
    }
  }, [isLoadingSubmissionCSV, submissionCSVDetails, getSubmissionCSVDetails]);

  const handleChange = (event: any, newValue: any) => {
    setValue(newValue);
  };

  if (!submissionCSVDetails || !submissionCSVDetails.data || submissionCSVDetails.data.length === 0) {
    return <CircularProgress data-testid="spinner" className="componentProgress" size={40} />;
  }

  return (
    <>
      <Tabs value={value} onChange={handleChange} aria-label="csv-groups" indicatorColor="primary">
        {submissionCSVDetails.data.map((dataItem: IGetSubmissionCSVForViewItem, dataItemIndex: number) => (
          <Tab key={dataItemIndex} label={dataItem.name} {...a11yProps(dataItemIndex)} />
        ))}
      </Tabs>
      <Box mt={2}>
        {submissionCSVDetails.data.map((dataItem: IGetSubmissionCSVForViewItem, dataItemIndex: number) => (
          <TabPanel key={dataItemIndex} value={value} index={dataItemIndex}>
            <TableContainer>
              <Table data-testid="submission-data-table" className={classes.table} aria-label="submission-data-table">
                <TableHead>
                  <TableRow>
                    {dataItem.headers.map((header: string, headerIndex: number) => (
                      <TableCell key={headerIndex} className={classes.heading}>
                        {header}
                      </TableCell>
                    ))}
                  </TableRow>
                </TableHead>
                <TableBody>
                  {dataItem.rows
                    .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                    .map((row: any[], rowIndex: number) => (
                      <TableRow key={rowIndex}>
                        {dataItem.headers.map((header: string, headerIndex: number) => (
                          <TableCell key={headerIndex}>{row[headerIndex]}</TableCell>
                        ))}
                      </TableRow>
                    ))}
                </TableBody>
              </Table>
            </TableContainer>
            <TablePagination
              rowsPerPageOptions={[25, 50, 75, 100]}
              component="div"
              count={dataItem.rows.length}
              rowsPerPage={rowsPerPage}
              page={page}
              onChangePage={(event: unknown, newPage: number) => handleChangePage(event, newPage, setPage)}
              onChangeRowsPerPage={(event: React.ChangeEvent<HTMLInputElement>) =>
                handleChangeRowsPerPage(event, setPage, setRowsPerPage)
              }
            />
          </TabPanel>
        ))}
      </Box>
    </>
  );
};

export default ObservationSubmissionCSV;
