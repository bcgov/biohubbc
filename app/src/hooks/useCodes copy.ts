import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { useBiohubApi } from './useBioHubApi';
import useDataLoader, { DataLoader } from './useDataLoader';

/**
 * Hook that fetches app code sets.
 *
 * @export
 * @return {*}  {DataLoader<IGetAllCodeSetsResponse>}
 */
export default function useCodes(): DataLoader<IGetAllCodeSetsResponse> {
  const api = useBiohubApi();

  return useDataLoader<IGetAllCodeSetsResponse>(api.codes.getAllCodeSets);
}
