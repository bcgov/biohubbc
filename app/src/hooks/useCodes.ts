import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { useEffect, useState } from 'react';
import { useBiohubApi } from './useBioHubApi';

export interface IUseCodes {
  codes: IGetAllCodeSetsResponse | undefined;
  isLoading: boolean;
  isReady: boolean;
}

/**
 * Hook that fetches app code sets.
 *
 * @export
 * @return {*}  {IUseCodes}
 */
export default function useCodes(): IUseCodes {
  const api = useBiohubApi();

  const [codes, setCodes] = useState<IGetAllCodeSetsResponse>();
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [isReady, setIsReady] = useState<boolean>(false);

  useEffect(() => {
    const fetchCodes = async () => {
      setIsLoading(true);

      const response = await api.codes.getAllCodeSets();

      setCodes(response);

      setIsReady(true);
    };

    if (codes) {
      return;
    }

    fetchCodes();
  }, [api.codes, codes]);

  return { codes, isLoading, isReady };
}
