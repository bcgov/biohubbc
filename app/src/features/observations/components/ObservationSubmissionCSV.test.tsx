import { cleanup, render, waitFor } from '@testing-library/react';
import { useBiohubApi } from 'hooks/useBioHubApi';
import React from 'react';
//import { MemoryRouter } from 'react-router';
import ObservationSubmissionCSV, { IObservationSubmissionCSVProps } from './ObservationSubmissionCSV';

jest.mock('../../../hooks/useBioHubApi');
const mockUseBiohubApi = {
  observation: {
    getSubmissionCSVForView: jest.fn()
  }
};

const mockBiohubApi = ((useBiohubApi as unknown) as jest.Mock<typeof mockUseBiohubApi>).mockReturnValue(
  mockUseBiohubApi
);

const renderContainer = (props: IObservationSubmissionCSVProps) => {
  return render(<ObservationSubmissionCSV {...props} />);
};

describe('ObservationSubmissionCSV', () => {
  beforeEach(() => {
    // clear mocks before each test
    mockBiohubApi().observation.getSubmissionCSVForView.mockClear();
  });

  afterEach(() => {
    cleanup();
  });

  it('renders a spinner with no data', async () => {
    const mockGetCSVData = jest.fn().mockResolvedValue(null);

    const { getByTestId } = renderContainer({ getCSVData: mockGetCSVData });

    await waitFor(() => {
      expect(getByTestId('spinner')).toBeInTheDocument();
    });
  });

  it('renders correctly with file csv data', async () => {
    const mockGetCSVData = mockBiohubApi().observation.getSubmissionCSVForView.mockResolvedValue({
      data: [
        {
          name: 'test 1',
          headers: ['header 1', 'header 2'],
          rows: [
            ['content 1', 'content 2'],
            ['content 3', 'content 4']
          ]
        },
        {
          name: 'test 2',
          headers: ['heading 1', 'heading 2'],
          rows: [
            ['awesome content 1', 'awesome content 2'],
            ['awesome content 3', 'awesome content 4']
          ]
        }
      ]
    });

    const { getByTestId } = renderContainer({ getCSVData: mockGetCSVData });

    await waitFor(() => {
      expect(getByTestId('submission-data-table')).toBeInTheDocument();
    });
  });
});
