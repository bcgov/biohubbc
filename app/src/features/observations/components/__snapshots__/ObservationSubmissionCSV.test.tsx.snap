// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationSubmissionCSV renders a spinner with no data 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root pageProgress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`ObservationSubmissionCSV renders correctly with file csv data 1`] = `
<DocumentFragment>
  <div
    class="MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed"
      style="overflow: hidden;"
    >
      <div
        aria-label="csv-groups"
        class="MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-controls="full-width-tabpanel-0"
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
          id="full-width-tab-0"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            test 1
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-controls="full-width-tabpanel-1"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          id="full-width-tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            test 2
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="PrivateTabIndicator-root-7 PrivateTabIndicator-colorPrimary-8 MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-11"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-labelledby="full-width-tab-0"
        id="full-width-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="MuiBox-root MuiBox-root-12"
        >
          <div
            class="MuiTypography-root MuiTypography-body1"
          >
            <div
              class="MuiTableContainer-root"
            >
              <table
                aria-label="submission-data-table"
                class="MuiTable-root makeStyles-table-4"
                data-testid="submission-data-table"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head makeStyles-heading-5"
                      scope="col"
                    >
                      header 1
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head makeStyles-heading-5"
                      scope="col"
                    >
                      header 2
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      content 1
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      content 2
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      content 3
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      content 4
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
              >
                <div
                  class="MuiTablePagination-spacer"
                />
                <p
                  class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                  id="mui-16053"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-16053 mui-35157"
                    class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                    id="mui-35157"
                    role="button"
                    tabindex="0"
                  >
                    50
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="50"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                >
                  1-2 of 2
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Previous page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    title="Previous page"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Next page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    title="Next page"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="full-width-tab-1"
        hidden=""
        id="full-width-tabpanel-1"
        role="tabpanel"
      />
    </div>
  </div>
</DocumentFragment>
`;
